src/validator.js

```diff
@@ -93,29 +104,59 @@ exports.validate = function(xmlData, options) {
             tagFound = true;
             //continue; //text may presents after self closing tag
           } else {
-            return isValid;
+            //the result from the nested function returns the position of the error within the attribute
+            //in order to get the 'true' error line, we need to add the calculate the position where the attribute begins (i - attrStr.length) and then add the position within the attribute
+            //this gives us the absolute index in the entire xml, which we can use to find the line at last
+            return {
+              err: {
+                code: isValid.err.code,
+                msg: isValid.err.msg,
+                line: getLineNumberForPosition(xmlData, i - attrStr.length + isValid.err.position),
+              }
+            };
           }
         } else if (closingTag) {
-          if(!result.tagClosed){
+          if (!result.tagClosed) {
             return {
-              err: {code: 'InvalidTag', msg: 'closing tag "' + tagName + "\" don't have proper closing."},
+              err: {
+                code: 'InvalidTag',
+                msg: 'closing tag "' + tagName + '" don\'t have proper closing.',
+                line: getLineNumberForPosition(xmlData, i),
+              },
             };
-          }else if (attrStr.trim().length > 0) {
+          } else if (attrStr.trim().length > 0) {
             return {
-              err: {code: 'InvalidTag', msg: 'closing tag "' + tagName + "\" can't have attributes or invalid starting."},
+              err: {
+                code: 'InvalidTag',
+                msg: 'closing tag "' + tagName + '" can\'t have attributes or invalid starting.',
+                line: getLineNumberForPosition(xmlData, i),
+              },
             };
           } else {
             const otg = tags.pop();
             if (tagName !== otg) {
               return {
-                err: {code: 'InvalidTag', msg: 'closing tag ' + otg + ' is expected inplace of ' + tagName + '.'},
+                err: {
+                  code: 'InvalidTag',
+                  msg: 'closing tag ' + otg + ' is expected inplace of ' + tagName + '.',
+                  line: getLineNumberForPosition(xmlData, i),
+                },
               };
             }
           }
         } else {
           const isValid = validateAttributeString(attrStr, options, regxAttrName);
           if (isValid !== true) {
-            return isValid;
+            //the result from the nested function returns the position of the error within the attribute
+            //in order to get the 'true' error line, we need to add the calculate the position where the attribute begins (i - attrStr.length) and then add the position within the attribute
```

Comment:
I believe you want to write "we need to calculate" instead of "we need to add the calculate"